src/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -1576,20 +1626,41 @@ public static BigInteger Pow(BigInteger value, Int32 exponent)
             left.AssertValid();
             right.AssertValid();
 
-            if (right.IsZero) return left;
-            if (left.IsZero) return right;
+            if (right._sign == 0)
+                return left;
+            if (left._sign < 0 != right._sign < 0)
+                return left - (-right);
```

Comment:
Isn't this going to result in an intermediary BigInteger being created, i.e. first a unary negation will create a BigInteger and then the subtraction will create another BigInteger?  That can't be avoided somehow?